app/src/main/java/com/example/dogbook/main/MainActivity.java

```diff
@@ -131,8 +132,13 @@ protected void onActivityResult(int requestCode, int resultCode, @Nullable Inten
         //A new post was composed
         if (requestCode == REQUEST_CODE_COMPOSE_ACTIVITY && resultCode == RESULT_OK) {
             navigationBar.setSelectedItemId(R.id.item_home);
-            //TODO: Update the main timeline
+            updateTimeline();
         }
     }
 
+    private void updateTimeline() {
+        TimelineFragment timelineFragment = (TimelineFragment) fragmentManager.findFragmentByTag("f" + ViewPagerAdapter.TIMELINE_PAGE);
+        timelineFragment.refreshPosts();
```

Comment:
Hm you could probably optimize this a little to not request all the posts over again but just try to update and get any newer posts the newest you have locally.  Then, just add those to the list, right?